packages/components/src/internal/components/navigation/ProductMenuSection.tsx

```diff
@@ -97,46 +84,30 @@ export const ProductMenuSection: FC<MenuSectionProps> = memo(props => {
         headerText
     );
 
-    // In order to make sure we don't break useRouteLeave we need to use <Link> for AppURLs and <a> for non-app URLs
-    let headerEl = label;
-    if (headerEl) {
+    let headerEl: ReactNode;
+    if (label) {
         const headerURL = config.useOriginalURL
             ? section.url
-            : createProductUrlFromPartsWithContainer(
-                  section.productId,
-                  currentProductId,
-                  containerPath,
-                  config.headerURLParams,
-                  config.headerURLPart ?? section.key
-              );
-
-        if (headerURL instanceof AppURL) {
-            headerEl = (
-                <Link to={headerURL.toString()} className="menu-section-link">
-                    {label}
-                </Link>
-            );
-        } else {
-            headerEl = <a href={getHref(headerURL)}>{label}</a>;
-        }
+            : AppURL.create(config.headerURLPart ?? section.key)
+                  .addParams(config.headerURLParams)
+                  .setContainerPath(containerPath)
+                  .setProductId(section.productId);
+
+        const className = headerURL instanceof AppURL ? 'menu-section-link' : undefined;
+        headerEl = (
+            <AppLink to={headerURL} className={className}>
+                {label}
+            </AppLink>
+        );
     }
 
     let emptyLink: ReactNode;
     if (config.emptyAppURL) {
-        const emptyURL = createProductUrl(section.productId, currentProductId, config.emptyAppURL, containerPath);
-        if (emptyURL instanceof AppURL) {
-            emptyLink = (
-                <Link className="menu-section-link" to={emptyURL.toString()}>
-                    {config.emptyURLText}
-                </Link>
-            );
-        } else {
-            emptyLink = (
-                <a className="menu-section-link" href={emptyURL}>
-                    {config.emptyURLText}
-                </a>
-            );
-        }
+        emptyLink = (
+            <AppLink to={config.emptyAppURL} className="menu-section-link">
```

Comment:
do we need to setContainerPath using `containerPath` in some way here?

Reply:
Yes, just pushed that fix, as well as fixes for the links in the product menu.